fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -1266,6 +1272,35 @@ public static FHIRSearchContext parseQueryParameters(String compartmentName, Str
         return parseQueryParameters(compartmentName, compartmentLogicalId, resourceType, queryParameters, true);
     }
 
+    /**
+     * Check the configuration to see if the flag enabling the compartment search
+     * optimization. Defaults to false so the behavior won't change unless it
+     * is explicitly enabled in fhir-server-config. This is important, because
+     * existing data must be reindexed (see $reindex custom operation) to
+     * generate values for the ibm-internal compartment relationship params.
+     * @return
+     */
+    public static boolean useStoredCompartmentParam() {
+        boolean result = false;
+        try {
+            String tenantId = FHIRRequestContext.get().getTenantId();
+            PropertyGroup fhirConfig = FHIRConfiguration.getInstance().loadConfigurationForTenant(tenantId);
+            if (fhirConfig == null) {
+                // fall back to default config (when unit tests don't provide config for a tenant)
+                fhirConfig = FHIRConfiguration.getInstance().loadConfiguration();
+            }
+
+            if (fhirConfig != null) {
+                result = fhirConfig.getBooleanProperty(FHIRConfiguration.PROPERTY_USE_STORED_COMPARTMENT_PARAM, false);
+            }
```

Comment:
I think this can all be replaced with FHIRConfigHelper.getBooleanProperty(FHIRConfiguration.PROPERTY_USE_STORED_COMPARTMENT_PARAM, false);

Reply:
done